vespa-hadoop/src/main/java/com/yahoo/vespa/hadoop/pig/VespaDocumentOperation.java

```diff
@@ -308,12 +387,38 @@ private static boolean shouldWritePartialUpdate(Operation op, int depth) {
     }
 
     private static void writePartialUpdate(Object value, Byte type, JsonGenerator g, String name, Properties properties, Schema schema, Operation op, int depth) throws IOException {
+        boolean isAssign = true;
+
         g.writeStartObject();
-        g.writeFieldName(PARTIAL_UPDATE_ASSIGN); // TODO: lookup field name in a property to determine correct operation
+        for (String label: operationMap.keySet()) {
+            if (properties.getProperty(label) != null) {
+                String[] p = properties.getProperty(label).split(",");
+                if (Arrays.asList(p).contains(name)) {
+                    g.writeFieldName(operationMap.get(label));
+                    isAssign = false;
+                }
+            }
+        }
+        if (isAssign) {
+            g.writeFieldName(PARTIAL_UPDATE_ASSIGN);
+        }
```

Comment:
Suggest to change the condition of writing assign operation.
There are two type of partial update: 1. single field partial update 2. whole document partial update
The 1st one is decided by property. The 2nd one is decided by operation name and the depth.
It's better to change the logic to the operation name and the depth so that we know this `assign` is added because of whole document partial update.